File: leetcode/0560-subarray-sum-equals-k/solution.py
class Solution(object):
    def subarraySum(self, nums, k):
        sums=defaultdict(int)
        prefix_sum=0
        res=0
        sums[0]=1
        for n in nums:
            prefix_sum+=n
            res+=sums[prefix_sum-k]
            sums[prefix_sum]+=1
        return res
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        

File: leetcode/0560-subarray-sum-equals-k/solution.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int sum=0;
        int ans=0;
        unordered_map<int,int> prefix;
        prefix[0]=1;
        for (int i=0;i<nums.size();i++){
            sum=sum+nums[i];
            if (prefix.find(sum-k)!=prefix.end()){
                ans=ans+prefix[sum-k];
            }
            prefix[sum]++;
        }
        return ans;
    }
};
